protocols/bftcosi/bftcosi.go

```diff
@@ -1,35 +1,51 @@
-// Package bftcosi store a novel way of scaling BFT for high scale internet
-// applications especially blockchains
 package bftcosi
 
+/*
+BFTCoSi is a byzantine-fault-tolerant protocol to sign a message given a
+verification-function. It uses two rounds of signing - the first round
+indicates the willingness of the rounds to sign the message, and the second
+round is only started if at least a 'threshold' number of nodes signed off in
+the first round.
+*/
+
 import (
-	"math"
+	"crypto/sha512"
 	"sync"
 
-	"github.com/dedis/cothority/cosi"
 	"github.com/dedis/cothority/log"
 	"github.com/dedis/cothority/sda"
 	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/cosi"
 )
 
 // VerificationFunction can be passes to each protocol node. It will be called
 // (in a go routine) during the (start/handle) challenge prepare phase of the
 // protocol. The passed message is the same as sent in the challenge phase.
-type VerificationFunction func(Msg []byte) bool
+type VerificationFunction func(Msg []byte, Data []byte) bool
```

Comment:
Maybe write a comment saying the `Data` part will be removed later when the `GenericConfig` works ? Because having a signature like this is definitely not something we want to keep. 
Either that or we make a PR for having `GenericConfig` working before merging this one.